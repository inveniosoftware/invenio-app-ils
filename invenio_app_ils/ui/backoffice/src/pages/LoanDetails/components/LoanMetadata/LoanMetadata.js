import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid, Segment, Form } from 'semantic-ui-react';

export class LoanMetadata extends Component {
  renderLoanMetadata(data) {
    return Object.keys(data.metadata).map(key => {
      return (
        <Form.Field key={key} name="loan-field">
          <label className="field-name">{key}</label>
          <p>{data.metadata[key]}</p>
        </Form.Field>
      );
    });
  }

  render() {
    let { data } = this.props;
    return (
      <Segment>
        <Grid>
          <Grid.Column>
            <h1>Loan - {data.metadata.loan_pid}</h1>
            <Form>{this.renderLoanMetadata(data)}</Form>
          </Grid.Column>
        </Grid>
      </Segment>
    );
  }
}

LoanMetadata.propTypes = {
  data: PropTypes.object.isRequired,
};
