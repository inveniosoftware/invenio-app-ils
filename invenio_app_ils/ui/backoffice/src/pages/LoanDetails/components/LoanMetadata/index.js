export { default as LoanMetadata } from './LoanMetadata';
