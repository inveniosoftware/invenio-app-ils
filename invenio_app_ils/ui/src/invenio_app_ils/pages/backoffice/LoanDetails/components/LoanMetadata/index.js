import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import LoanMetadataComponent from './LoanMetadata';

const mapStateToProps = state => ({
  error: state.loanDetails.error,
  loanDetails: state.loanDetails.data,
});

export const LoanMetadata = compose(
  withRouter,
  connect(mapStateToProps)
)(LoanMetadataComponent);
