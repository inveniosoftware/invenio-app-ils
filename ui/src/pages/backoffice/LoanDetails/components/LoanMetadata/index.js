import { connect } from 'react-redux';

import LoanMetadataComponent from './LoanMetadata';

const mapStateToProps = state => ({
  error: state.loanDetails.error,
  loanDetails: state.loanDetails.data,
});

export const LoanMetadata = connect(mapStateToProps)(LoanMetadataComponent);
