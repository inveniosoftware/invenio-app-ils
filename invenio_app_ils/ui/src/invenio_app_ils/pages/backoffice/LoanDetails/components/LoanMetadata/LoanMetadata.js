import { invenioConfig } from '../../../../../common/config';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Divider,
  Grid,
  Header,
  Icon,
  Message,
  Segment,
} from 'semantic-ui-react';
import { ItemMetadata } from '../ItemMetadata';
import { LoanActions } from '../LoanActions';
import { AvailableItems } from '../AvailableItems';
import { MetadataTable } from '../../../components/MetadataTable';
import isEmpty from 'lodash/isEmpty';
import { toShortDateTime } from '../../../../../common/api/date';

export default class LoanMetadata extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isAvailableItemsVisible: isEmpty(this.props.loanDetails.metadata.item),
    };
  }

  showAvailableItems = (newState = true) => {
    this.setState({ isAvailableItemsVisible: newState });
  };

  isItemDeleted = () => {
    const data = this.props.loanDetails;
    return isEmpty(data.metadata.item) && data.metadata.item_pid !== '';
  };

  renderItem() {
    const data = this.props.loanDetails;
    return (
      !isEmpty(data.metadata.item) && (
        <ItemMetadata
          item={data.metadata.item}
          loanState={data.metadata.state}
          changeItemClickHandler={this.showAvailableItems}
        />
      )
    );
  }

  renderDeletedItem() {
    const data = this.props.loanDetails;
    return (
      this.isItemDeleted() && (
        <Message
          warning
          icon="trash alternate"
          header={`The Item ${
            data.metadata.item_pid
          } assigned to this loan has been deleted!`}
          content={[
            'If you need further assistance contact ',
            <a
              href={`mailto:${
                invenioConfig.support_email
              }?subject=Deleted Item ${data.metadata.item_pid}`}
              key="support-email"
            >
              <Icon name="mail" />
              {invenioConfig.support_email}
            </a>,
          ]}
        />
      )
    );
  }

  prepareLeftData(data) {
    const rows = [
      { name: 'Document pid', value: data.metadata.document_pid },
      { name: 'Item pid', value: data.metadata.item_pid },
      { name: 'Patron pid', value: data.metadata.patron_pid },
      { name: 'Pickup Location pid', value: data.metadata.pickup_location_pid },
      {
        name: 'Transaction Location pid',
        value: data.metadata.transaction_location_pid,
      },
      { name: 'State', value: data.metadata.state },
    ];
    return rows;
  }

  prepareRightData(data) {
    const { cancel_reason: reason, state } = data.metadata;
    const rows = [
      {
        name: 'Transaction date',
        value: toShortDateTime(data.metadata.transaction_date),
      },
      {
        name: 'Expire Date',
        value: toShortDateTime(data.metadata.request_expire_date),
      },
    ];
    if (state === 'CANCELLED' && !isEmpty(reason)) {
      rows.push({
        name: 'Cancel Reason',
        value: reason,
      });
    }
    return rows;
  }

  renderLoan() {
    const data = this.props.loanDetails;
    const leftRows = this.prepareLeftData(data);
    const rightRows = this.prepareRightData(data);
    return (
      <Grid padded columns={2}>
        <Grid.Column width={16}>
          <Header as="h1">Loan - {data.pid}</Header>
        </Grid.Column>
        <Grid.Column>
          <MetadataTable rows={leftRows} />
        </Grid.Column>
        <Grid.Column>
          <MetadataTable rows={rightRows} />
        </Grid.Column>
      </Grid>
    );
  }

  renderAvailableItems() {
    return (
      this.state.isAvailableItemsVisible && (
        <>
          <Divider />
          <AvailableItems loan={this.props.loanDetails} />
        </>
      )
    );
  }

  render() {
    return (
      <Segment>
        {this.renderLoan()}
        <Divider />
        <LoanActions />
        <Divider />
        {this.renderItem()}
        {this.renderDeletedItem()}
        {this.renderAvailableItems()}
      </Segment>
    );
  }
}

LoanMetadata.propTypes = {
  loanDetails: PropTypes.object.isRequired,
};
