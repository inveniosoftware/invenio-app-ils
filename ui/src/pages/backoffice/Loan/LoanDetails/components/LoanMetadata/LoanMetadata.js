import { invenioConfig } from '@config';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Divider, Grid, Header, Message, Segment } from 'semantic-ui-react';
import { ItemMetadata } from '../ItemMetadata';
import { LoanActions } from '../LoanActions';
import { AvailableItems } from '../AvailableItems';
import {
  OverdueLoanSendMailModal,
  MetadataTable,
} from '@pages/backoffice/components';
import isEmpty from 'lodash/isEmpty';
import { toShortDateTime } from '@api/date';
import { LoanLinkToItem } from '../../../../components/Loan';

export default class LoanMetadata extends Component {
  constructor(props) {
    super(props);
    const isRequest = invenioConfig.circulation.loanRequestStates.includes(
      this.props.loanDetails.metadata.state
    );
    this.state = {
      isAvailableItemsVisible: isRequest,
    };
  }

  showAvailableItems = (newState = true) => {
    this.setState({ isAvailableItemsVisible: newState });
  };

  isItemDeleted = () => {
    const data = this.props.loanDetails;
    return isEmpty(data.metadata.item) && data.metadata.item_pid;
  };

  renderItem() {
    const data = this.props.loanDetails;
    return (
      !isEmpty(data.metadata.item) && (
        <ItemMetadata
          item={data.metadata.item}
          loanState={data.metadata.state}
          changeItemClickHandler={this.showAvailableItems}
        />
      )
    );
  }

  renderDeletedItem() {
    const data = this.props.loanDetails;
    return (
      this.isItemDeleted() && (
        <Message
          warning
          icon="trash alternate"
          header={`Item deleted`}
          content={`The item ${data.metadata.item_pid.value} assigned to this loan has been deleted.`}
        />
      )
    );
  }

  prepareLeftData(data) {
    const rows = [
      { name: 'Document pid', value: data.metadata.document_pid },
      {
        name: 'Item pid',
        value: data.metadata.item_pid ? (
          <LoanLinkToItem itemPid={data.metadata.item_pid}>
            {data.metadata.item_pid.value}
          </LoanLinkToItem>
        ) : (
          '-'
        ),
      },
      { name: 'Patron pid', value: data.metadata.patron_pid },
      { name: 'Pickup Location pid', value: data.metadata.pickup_location_pid },
      { name: 'State', value: data.metadata.state },
    ];
    return rows;
  }

  getDelivery(delivery) {
    if (delivery && 'method' in delivery) {
      return invenioConfig.circulation.deliveryMethods[delivery.method];
    }
    return 'NOT PROVIDED';
  }

  prepareRightData(data) {
    const { cancel_reason: reason, state } = data.metadata;
    const rows = [
      {
        name: 'Transaction date',
        value: toShortDateTime(data.metadata.transaction_date),
      },
      {
        name: 'Delivery',
        value: this.getDelivery(data.metadata.delivery),
      },
    ];
    if (state === 'CANCELLED' && !isEmpty(reason)) {
      rows.push({
        name: 'Cancel Reason',
        value: reason,
      });
    }
    return rows;
  }

  renderLoan() {
    const data = this.props.loanDetails;
    const leftRows = this.prepareLeftData(data);
    const rightRows = this.prepareRightData(data);
    return (
      <Grid padded columns={2}>
        <Grid.Column width={16}>
          <Header as="h1">Loan - {data.pid}</Header>
        </Grid.Column>
        <Grid.Column>
          <MetadataTable rows={leftRows} />
        </Grid.Column>
        <Grid.Column>
          <MetadataTable rows={rightRows} />
          {this.renderMailButton()}
        </Grid.Column>
      </Grid>
    );
  }

  renderMailButton() {
    const loan = this.props.loanDetails;
    return loan.metadata.is_overdue && <OverdueLoanSendMailModal loan={loan} />;
  }

  renderAvailableItems() {
    return (
      this.state.isAvailableItemsVisible && (
        <>
          {this.isItemDeleted() && <Divider />}
          <AvailableItems loan={this.props.loanDetails} />
        </>
      )
    );
  }

  render() {
    return (
      <Segment>
        {this.renderLoan()}
        <Divider />
        <LoanActions />
        <Divider />
        {this.renderItem()}
        {this.renderDeletedItem()}
        {this.renderAvailableItems()}
      </Segment>
    );
  }
}

LoanMetadata.propTypes = {
  loanDetails: PropTypes.object.isRequired,
};
