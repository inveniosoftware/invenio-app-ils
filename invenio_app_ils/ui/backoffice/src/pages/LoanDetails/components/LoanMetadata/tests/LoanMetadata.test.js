import React from 'react';
import { shallow } from 'enzyme';
import LoanMetadata from '../LoanMetadata';

describe('LoanMetadata component', () => {
  it('renders initial state', () => {
    const data = {
      metadata: {},
    };
    const component = shallow(<LoanMetadata data={data} />);
    expect(component).toMatchSnapshot();
  });

  it('renders data', () => {
    const data = {
      metadata: {
        title: 'title',
        author: 'author',
      },
    };
    const component = shallow(<LoanMetadata data={data} />);
    expect(component).toMatchSnapshot();
  });
});
