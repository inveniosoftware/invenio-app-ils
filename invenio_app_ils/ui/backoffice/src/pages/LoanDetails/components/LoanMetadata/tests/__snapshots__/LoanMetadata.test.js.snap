// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanMetadata component renders data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoanMetadata
    data={
      Object {
        "metadata": Object {
          "author": "author",
          "title": "title",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
        >
          Loan Information
        </Header>,
        <Table
          as="table"
          celled={true}
          compact={true}
          definition={true}
          textAlign="center"
        >
          <TableHeader
            as="thead"
            fullWidth={true}
          >
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableHeaderCell
                as="th"
              >
                Property
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Value
              </TableHeaderCell>
            </TableRow>
          </TableHeader>
          <TableBody
            as="tbody"
          >
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableCell
                as="td"
                collapsing={true}
              >
                title
              </TableCell>
              <TableCell
                as="td"
              >
                title
              </TableCell>
            </TableRow>
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableCell
                as="td"
                collapsing={true}
              >
                author
              </TableCell>
              <TableCell
                as="td"
              >
                author
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": "Loan Information",
        },
        "ref": null,
        "rendered": "Loan Information",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "table",
          "celled": true,
          "children": Array [
            <TableHeader
              as="thead"
              fullWidth={true}
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Property
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Value
                </TableHeaderCell>
              </TableRow>
            </TableHeader>,
            <TableBody
              as="tbody"
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableCell
                  as="td"
                  collapsing={true}
                >
                  title
                </TableCell>
                <TableCell
                  as="td"
                >
                  title
                </TableCell>
              </TableRow>
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableCell
                  as="td"
                  collapsing={true}
                >
                  author
                </TableCell>
                <TableCell
                  as="td"
                >
                  author
                </TableCell>
              </TableRow>
            </TableBody>,
          ],
          "compact": true,
          "definition": true,
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "thead",
              "children": <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Property
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Value
                </TableHeaderCell>
              </TableRow>,
              "fullWidth": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tr",
                "cellAs": "td",
                "children": Array [
                  <TableHeaderCell
                    as="th"
                  >
                    Property
                  </TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
                  >
                    Value
                  </TableHeaderCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Property",
                  },
                  "ref": null,
                  "rendered": "Property",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Value",
                  },
                  "ref": null,
                  "rendered": "Value",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tbody",
              "children": Array [
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableCell
                    as="td"
                    collapsing={true}
                  >
                    title
                  </TableCell>
                  <TableCell
                    as="td"
                  >
                    title
                  </TableCell>
                </TableRow>,
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableCell
                    as="td"
                    collapsing={true}
                  >
                    author
                  </TableCell>
                  <TableCell
                    as="td"
                  >
                    author
                  </TableCell>
                </TableRow>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "title",
                "nodeType": "function",
                "props": Object {
                  "as": "tr",
                  "cellAs": "td",
                  "children": Array [
                    <TableCell
                      as="td"
                      collapsing={true}
                    >
                      title
                    </TableCell>,
                    <TableCell
                      as="td"
                    >
                      title
                    </TableCell>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "td",
                      "children": "title",
                      "collapsing": true,
                    },
                    "ref": null,
                    "rendered": "title",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "td",
                      "children": "title",
                    },
                    "ref": null,
                    "rendered": "title",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "author",
                "nodeType": "function",
                "props": Object {
                  "as": "tr",
                  "cellAs": "td",
                  "children": Array [
                    <TableCell
                      as="td"
                      collapsing={true}
                    >
                      author
                    </TableCell>,
                    <TableCell
                      as="td"
                    >
                      author
                    </TableCell>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "td",
                      "children": "author",
                      "collapsing": true,
                    },
                    "ref": null,
                    "rendered": "author",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "td",
                      "children": "author",
                    },
                    "ref": null,
                    "rendered": "author",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
          >
            Loan Information
          </Header>,
          <Table
            as="table"
            celled={true}
            compact={true}
            definition={true}
            textAlign="center"
          >
            <TableHeader
              as="thead"
              fullWidth={true}
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Property
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Value
                </TableHeaderCell>
              </TableRow>
            </TableHeader>
            <TableBody
              as="tbody"
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableCell
                  as="td"
                  collapsing={true}
                >
                  title
                </TableCell>
                <TableCell
                  as="td"
                >
                  title
                </TableCell>
              </TableRow>
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableCell
                  as="td"
                  collapsing={true}
                >
                  author
                </TableCell>
                <TableCell
                  as="td"
                >
                  author
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": "Loan Information",
          },
          "ref": null,
          "rendered": "Loan Information",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "table",
            "celled": true,
            "children": Array [
              <TableHeader
                as="thead"
                fullWidth={true}
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableHeaderCell
                    as="th"
                  >
                    Property
                  </TableHeaderCell>
                  <TableHeaderCell
                    as="th"
                  >
                    Value
                  </TableHeaderCell>
                </TableRow>
              </TableHeader>,
              <TableBody
                as="tbody"
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableCell
                    as="td"
                    collapsing={true}
                  >
                    title
                  </TableCell>
                  <TableCell
                    as="td"
                  >
                    title
                  </TableCell>
                </TableRow>
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableCell
                    as="td"
                    collapsing={true}
                  >
                    author
                  </TableCell>
                  <TableCell
                    as="td"
                  >
                    author
                  </TableCell>
                </TableRow>
              </TableBody>,
            ],
            "compact": true,
            "definition": true,
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "thead",
                "children": <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableHeaderCell
                    as="th"
                  >
                    Property
                  </TableHeaderCell>
                  <TableHeaderCell
                    as="th"
                  >
                    Value
                  </TableHeaderCell>
                </TableRow>,
                "fullWidth": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "tr",
                  "cellAs": "td",
                  "children": Array [
                    <TableHeaderCell
                      as="th"
                    >
                      Property
                    </TableHeaderCell>,
                    <TableHeaderCell
                      as="th"
                    >
                      Value
                    </TableHeaderCell>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "th",
                      "children": "Property",
                    },
                    "ref": null,
                    "rendered": "Property",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "th",
                      "children": "Value",
                    },
                    "ref": null,
                    "rendered": "Value",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tbody",
                "children": Array [
                  <TableRow
                    as="tr"
                    cellAs="td"
                  >
                    <TableCell
                      as="td"
                      collapsing={true}
                    >
                      title
                    </TableCell>
                    <TableCell
                      as="td"
                    >
                      title
                    </TableCell>
                  </TableRow>,
                  <TableRow
                    as="tr"
                    cellAs="td"
                  >
                    <TableCell
                      as="td"
                      collapsing={true}
                    >
                      author
                    </TableCell>
                    <TableCell
                      as="td"
                    >
                      author
                    </TableCell>
                  </TableRow>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "title",
                  "nodeType": "function",
                  "props": Object {
                    "as": "tr",
                    "cellAs": "td",
                    "children": Array [
                      <TableCell
                        as="td"
                        collapsing={true}
                      >
                        title
                      </TableCell>,
                      <TableCell
                        as="td"
                      >
                        title
                      </TableCell>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": "td",
                        "children": "title",
                        "collapsing": true,
                      },
                      "ref": null,
                      "rendered": "title",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": "td",
                        "children": "title",
                      },
                      "ref": null,
                      "rendered": "title",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "author",
                  "nodeType": "function",
                  "props": Object {
                    "as": "tr",
                    "cellAs": "td",
                    "children": Array [
                      <TableCell
                        as="td"
                        collapsing={true}
                      >
                        author
                      </TableCell>,
                      <TableCell
                        as="td"
                      >
                        author
                      </TableCell>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": "td",
                        "children": "author",
                        "collapsing": true,
                      },
                      "ref": null,
                      "rendered": "author",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": "td",
                        "children": "author",
                      },
                      "ref": null,
                      "rendered": "author",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`LoanMetadata component renders initial state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoanMetadata
    data={
      Object {
        "metadata": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
        >
          Loan Information
        </Header>,
        <Table
          as="table"
          celled={true}
          compact={true}
          definition={true}
          textAlign="center"
        >
          <TableHeader
            as="thead"
            fullWidth={true}
          >
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableHeaderCell
                as="th"
              >
                Property
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Value
              </TableHeaderCell>
            </TableRow>
          </TableHeader>
          <TableBody
            as="tbody"
          />
        </Table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": "Loan Information",
        },
        "ref": null,
        "rendered": "Loan Information",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "table",
          "celled": true,
          "children": Array [
            <TableHeader
              as="thead"
              fullWidth={true}
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Property
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Value
                </TableHeaderCell>
              </TableRow>
            </TableHeader>,
            <TableBody
              as="tbody"
            />,
          ],
          "compact": true,
          "definition": true,
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "thead",
              "children": <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Property
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Value
                </TableHeaderCell>
              </TableRow>,
              "fullWidth": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tr",
                "cellAs": "td",
                "children": Array [
                  <TableHeaderCell
                    as="th"
                  >
                    Property
                  </TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
                  >
                    Value
                  </TableHeaderCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Property",
                  },
                  "ref": null,
                  "rendered": "Property",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Value",
                  },
                  "ref": null,
                  "rendered": "Value",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tbody",
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
          >
            Loan Information
          </Header>,
          <Table
            as="table"
            celled={true}
            compact={true}
            definition={true}
            textAlign="center"
          >
            <TableHeader
              as="thead"
              fullWidth={true}
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Property
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Value
                </TableHeaderCell>
              </TableRow>
            </TableHeader>
            <TableBody
              as="tbody"
            />
          </Table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": "Loan Information",
          },
          "ref": null,
          "rendered": "Loan Information",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "table",
            "celled": true,
            "children": Array [
              <TableHeader
                as="thead"
                fullWidth={true}
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableHeaderCell
                    as="th"
                  >
                    Property
                  </TableHeaderCell>
                  <TableHeaderCell
                    as="th"
                  >
                    Value
                  </TableHeaderCell>
                </TableRow>
              </TableHeader>,
              <TableBody
                as="tbody"
              />,
            ],
            "compact": true,
            "definition": true,
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "thead",
                "children": <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableHeaderCell
                    as="th"
                  >
                    Property
                  </TableHeaderCell>
                  <TableHeaderCell
                    as="th"
                  >
                    Value
                  </TableHeaderCell>
                </TableRow>,
                "fullWidth": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "tr",
                  "cellAs": "td",
                  "children": Array [
                    <TableHeaderCell
                      as="th"
                    >
                      Property
                    </TableHeaderCell>,
                    <TableHeaderCell
                      as="th"
                    >
                      Value
                    </TableHeaderCell>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "th",
                      "children": "Property",
                    },
                    "ref": null,
                    "rendered": "Property",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "th",
                      "children": "Value",
                    },
                    "ref": null,
                    "rendered": "Value",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tbody",
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
